src/node.cc

```diff
@@ -4556,7 +4557,9 @@ inline int Start(Isolate* isolate, IsolateData* isolate_data,
   {
     SealHandleScope seal(isolate);
     bool more;
+    PERFORMANCE_MARK(&env, LOOP_START);
     do {
+      PERFORMANCE_MARK(&env, LOOP_FRAME);
```

Comment:
it adds a uv_idle_t, uv_check_t and uv_prepare_t to the loop to capture the current hrtime for each. It's not perfect by any means given that other handles can execute before them. What would be more ideal would be a change to libuv to capture those timestamps for us at the start of each phase, then provide an API to extract those.